### PR TITLE
Proposal for CONTRIBUTING.md

### PR DESCRIPTION
**Pull request by jackzampolin**
I've taken what's in `cosmos-sdk` and updated for here. Would love comments feedback!

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+We use [`dep`](https://github.com/golang/dep) to manage dependencies.
+
+That said, the master branch of every Cosmos repository should just build
+with `go get`, which means they should be kept up-to-date with their
+dependencies so we can get away with telling people they can just `go get` our
+software.
+
+Since some dependencies are not under our control, a third party may break our
+build, in which case we can fall back on `dep ensure` (or `make
+get_vendor_deps`). Even for dependencies under our control, `dep` helps us to
```
Comment (alexanderbez) [inline CONTRIBUTING.md]: I'll be changing this to `get-vendor-deps` in ethermint 👍

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+includes its continuous integration status using a badge in the `README.md`.
+
+## Branching Model and Release
+
+User-facing repos should adhere to the branching model: http://nvie.com/posts/a-successful-git-branching-model/.
+That is, these repos should be well versioned, and any merge to master requires a version bump and tagged release.
+
+Libraries need not follow the model strictly, but would be wise to.
+
+The SDK utilizes [semantic versioning](https://semver.org/).
```
Comment (alexanderbez) [inline CONTRIBUTING.md]: Ethermint*?

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -0,0 +1,105 @@
...
+
+### Development Procedure:
+- the latest state of development is on `develop`
+- `develop` must never fail `make test`
+- `develop` should not fail `make test_lint`
```
Comment (alexanderbez) [inline CONTRIBUTING.md]: I'll be changing this to `test-lint`.

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -0,0 +1,105 @@
...
+- no --force onto `develop` (except when reverting a broken commit, which should seldom happen)
+- create a development branch either on github.com/cosmos/cosmos-sdk, or your fork (using `git remote add origin`)
```
Comment (alexanderbez) [inline CONTRIBUTING.md]: ethermint*

File: CONTRIBUTING.md [hunk 5]
```diff
@@ -0,0 +1,105 @@
...
+- before submitting a pull request, begin `git rebase` on top of `develop`
```
Comment (alexanderbez) [inline CONTRIBUTING.md]: I'd add a note here suggesting it'd be wise to squash commits before rebasing.

File: CONTRIBUTING.md [hunk 6]
```diff
@@ -0,0 +1,105 @@
...
+
+### Pull Merge Procedure:
+- ensure pull branch is rebased on develop
```
Comment (alexanderbez) [inline CONTRIBUTING.md]: I'd also add a note here stating commits should be squashed.

File: CONTRIBUTING.md [hunk 7]
```diff
@@ -0,0 +1,105 @@
...
+- run `make test` and `make test_cli` to ensure that all tests pass
```
Comment (alexanderbez) [inline CONTRIBUTING.md]: `test-cli`

File: CONTRIBUTING.md [hunk 8]
```diff
@@ -0,0 +1,105 @@
...
+- merge pull request
+- push master may request that pull requests be rebased on top of `unstable`
```
Comment (alexanderbez) [inline CONTRIBUTING.md]: I'd remove this line for now 👍

**Review by alexanderbez — CHANGES_REQUESTED** [review]
Thanks for this @jackzampolin. I left some minor remarks :-)

File: CONTRIBUTING.md (hunk 1 above)
Comment (jackzampolin) [reply CONTRIBUTING.md]: Awesome changed here.

File: CONTRIBUTING.md (hunk 2 above)
Comment (jackzampolin) [reply CONTRIBUTING.md]: 🙊

File: CONTRIBUTING.md (hunk 3 above)
Comment (jackzampolin) [reply CONTRIBUTING.md]: Awesome changed here.

File: CONTRIBUTING.md (hunk 4 above)
Comment (jackzampolin) [reply CONTRIBUTING.md]: Got this one and a couple of others too. Thanks for the 👀

File: CONTRIBUTING.md (hunk 5 above)
Comment (jackzampolin) [reply CONTRIBUTING.md]: Done, also added link to procedure

File: CONTRIBUTING.md (hunk 6 above)
Comment (jackzampolin) [reply CONTRIBUTING.md]: Done, also added link to procedure

File: CONTRIBUTING.md (hunk 7 above)
Comment (jackzampolin) [reply CONTRIBUTING.md]: Awesome changed here.

File: CONTRIBUTING.md (hunk 8 above)
Comment (jackzampolin) [reply CONTRIBUTING.md]: 👍

**Comment by jackzampolin** [comment]
Thanks for the comments! Addressed them!

**Comment by alexanderbez** [comment]
@jackzampolin mind resolving the merge conflicts?

**Pull request merged**